From f6f51bd16093383812139c84183374d6c39a02ac Mon Sep 17 00:00:00 2001
From: kevall474 <kevall474@tuta.io>
Date: Sat, 12 Jun 2021 15:08:56 +0000
Subject: [PATCH] Makefile Turn off loop vectorization for GCC O3 opti

---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index 8b8dbbf23..fce7b19e7 100644
--- a/Makefile
+++ b/Makefile
@@ -807,6 +807,9 @@ KBUILD_RUSTCFLAGS_OPT_LEVEL_MAP := 2
 else ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE_O3
 KBUILD_CFLAGS += -O3
 KBUILD_RUSTCFLAGS_OPT_LEVEL_MAP := 3
+ifdef CONFIG_CC_IS_GCC
+KBUILD_CFLAGS += -fno-tree-loop-vectorize
+endif
 else ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
 KBUILD_CFLAGS += -Os
 KBUILD_RUSTCFLAGS_OPT_LEVEL_MAP := z
-- 
2.31.1

