From 361168fc825661dc74083cbd8426f6590bc8e84c Mon Sep 17 00:00:00 2001
From: kevall474 <kevall474@tuta.io>
Date: Sun, 23 May 2021 11:33:59 +0000
Subject: [PATCH] Remove asm fr more optimization flags patche

---
 include/linux/compiler-gcc.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/compiler-gcc.h b/include/linux/compiler-gcc.h
index 5d97ef738a57..77491b34dc8a 100644
--- a/include/linux/compiler-gcc.h
+++ b/include/linux/compiler-gcc.h
@@ -57,7 +57,7 @@
  *
  * Adding an empty inline assembly before it works around the problem
  */
-#define barrier_before_unreachable() asm volatile("")
+/*#define barrier_before_unreachable() asm volatile("")*/
 
 /*
  * Mark a position in code as unreachable.  This can be used to
@@ -88,7 +88,7 @@
  *
  * (asm goto is automatically volatile - the naming reflects this.)
  */
-#define asm_volatile_goto(x...)	do { asm goto(x); asm (""); } while (0)
+/*#define asm_volatile_goto(x...)	do { asm goto(x); asm (""); } while (0)*/
 
 #if defined(CONFIG_ARCH_USE_BUILTIN_BSWAP)
 #define __HAVE_BUILTIN_BSWAP32__
-- 
2.31.1

