From c7a9e4e0b7bfda7033afc446cc94f01238763042 Mon Sep 17 00:00:00 2001
From: kevall474 <kevall474@tuta.io>
Date: Sun, 6 Jun 2021 15:40:42 +0000
Subject: [PATCH] init Kconfig add O1 flag

---
 Makefile     | 3 +++
 init/Kconfig | 6 ++++++
 2 files changed, 9 insertions(+)

diff --git a/Makefile b/Makefile
index ba0b6326d..9c02808b5 100644
--- a/Makefile
+++ b/Makefile
@@ -810,6 +810,9 @@ KBUILD_RUSTCFLAGS_OPT_LEVEL_MAP := 3
 else ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
 KBUILD_CFLAGS += -Os
 KBUILD_RUSTCFLAGS_OPT_LEVEL_MAP := z
+else ifdef CONFIG_CC_OPTIMIZE_BASAL
+KBUILD_CFLAGS += -O1
+KBUILD_RUSTCFLAGS_OPT_LEVEL_MAP := 1
 endif
 
 ifdef CONFIG_RUST_OPT_LEVEL_SIMILAR_AS_CHOSEN_FOR_C
diff --git a/init/Kconfig b/init/Kconfig
index 173a47401..37c2effbc 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -1334,6 +1334,12 @@ choice
 	prompt "Compiler optimization level"
 	default CC_OPTIMIZE_FOR_PERFORMANCE
 
+config CC_OPTIMIZE_BASAL
+	bool "Optimize for performance (-O1)"
+	help
+	  Choosing this option will pass "-O1" to your compiler resulting
+	  in basal optimization, possibly speeding up compilation.
+
 config CC_OPTIMIZE_FOR_PERFORMANCE
 	bool "Optimize for performance (-O2)"
 	help
-- 
2.31.1

