From c226c4e3a58062e7030796a8972da5c070d0a497 Mon Sep 17 00:00:00 2001
From: kevall474 <kevall474@tuta.io>
Date: Sat, 12 Dec 2020 20:00:46 -0500
Subject: [PATCH] makefile add more optimize flags

---
 Makefile | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0983973bc..4b805a602 100644
--- a/Makefile
+++ b/Makefile
@@ -733,12 +733,22 @@ KBUILD_CFLAGS	+= $(call cc-disable-warning, format-truncation)
 KBUILD_CFLAGS	+= $(call cc-disable-warning, format-overflow)
 KBUILD_CFLAGS	+= $(call cc-disable-warning, address-of-packed-member)
 
-ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE
+ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE_O0
+KBUILD_CFLAGS += -O0
+else ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE_O
+KBUILD_CFLAGS += -O
+else ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE_O1
+KBUILD_CFLAGS += -O1
+else ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE_O2
 KBUILD_CFLAGS += -O2
 else ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE_O3
 KBUILD_CFLAGS += -O3
 else ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
 KBUILD_CFLAGS += -Os
+else ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE_Ofast
+KBUILD_CFLAGS += -Ofast
+else ifdef CONFIG_CC_OPTIMIZE_FOR_DEBUGGING
+KBUILD_CFLAGS += -Og
 endif
 
 # Tell gcc to never replace conditional load with a non-conditional one
-- 
2.29.2

