From: kevall474
Date: 2020-11-17
Subject: glitched - MuQSS from Tk-Glitch <ti3nou@gmail.com> rebase by kevall474 <kevall474@tuta.io>

diff --git a/kernel/sched/MuQSS.c b/kernel/sched/MuQSS.c
index 84a1d08d68551..57c3036a68952 100644
--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -163,7 +167,11 @@ int sched_interactive __read_mostly = 1;
  * are allowed to run five seconds as real time tasks. This is the total over
  * all online cpus.
  */
+#ifdef CONFIG_ZENIFY
+int sched_iso_cpu __read_mostly = 25;
+#else
 int sched_iso_cpu __read_mostly = 70;
+#endif

 /*
  * sched_yield_type - Choose what sort of yield sched_yield will perform.

diff --git a/scripts/headers_install.sh b/scripts/headers_install.sh
index dd554bd43..75030ad93 100755
--- a/scripts/headers_install.sh
+++ b/scripts/headers_install.sh
@@ -89,6 +89,7 @@ include/uapi/linux/atmdev.h:CONFIG_COMPAT
 include/uapi/linux/eventpoll.h:CONFIG_PM_SLEEP
 include/uapi/linux/hw_breakpoint.h:CONFIG_HAVE_MIXED_BREAKPOINTS_REGS
 include/uapi/linux/pktcdvd.h:CONFIG_CDROM_PKTCDVD_WCACHE
 +include/uapi/linux/sched.h:CONFIG_SCHED_MUQSS
  "

for c in $configs
