From: kevall474
Date: 2020-11-17
Subject: glitched - MuQSS from Tk-Glitch <ti3nou@gmail.com> rebase by kevall474 <kevall474@tuta.io>

diff --git a/kernel/sched/MuQSS.c b/kernel/sched/MuQSS.c
index 84a1d08d68551..57c3036a68952 100644
--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -163,7 +167,11 @@ int sched_interactive __read_mostly = 1;
  * are allowed to run five seconds as real time tasks. This is the total over
  * all online cpus.
  */
+#ifdef CONFIG_ZENIFY
+int sched_iso_cpu __read_mostly = 25;
+#else
 int sched_iso_cpu __read_mostly = 70;
+#endif

 /*
  * sched_yield_type - Choose what sort of yield sched_yield will perform.
